src/utils/registry.ts

```diff
@@ -13,18 +22,26 @@ import { GithubRemote } from './githubApi';
  * @param packageDirPath The package directory.
  * @param version The package version.
  */
-export function getPackageManifest(
+export async function getPackageManifest(
   baseDir: string,
-  packageDirPath: string,
+  type: RegistryPackageType,
+  canonicalName: string,
   version: string
-): any {
+): Promise<{ versionFilePath: string; packageManifest: any }> {
+  const packageDirPath = getPackageDirPath(type, canonicalName);
   const versionFilePath = path.join(baseDir, packageDirPath, `${version}.json`);
-  if (fs.existsSync(versionFilePath)) {
+  if (existsSync(versionFilePath)) {
     reportError(`Version file for "${version}" already exists. Aborting.`);
   }
   const packageManifestPath = path.join(baseDir, packageDirPath, 'latest.json');
-  logger.debug('Reading the current configuration from "latest.json"...');
-  return JSON.parse(fs.readFileSync(packageManifestPath).toString()) || {};
+  logger.debug('Reading the current configuration from', packageManifestPath);
+  return {
+    versionFilePath,
+    packageManifest:
+      JSON.parse(
+        await fsPromises.readFile(packageManifestPath, { encoding: 'utf-8' })
```

Comment:
For my own learning: this is equivalent to using `readFileSync`, yes?

Reply:
Codewise, yes. Runtime, it should be slightly slower as it adds async cycles but should allow other things to happen in the meantime (like other I/O).